File: shoppingcart/src/main/java/com/deloitte/shoppingcart/service/UserService.java
package com.deloitte.shoppingcart.service;

import java.util.List;
import java.util.Optional;

import com.deloitte.shoppingcart.model.User;


public interface UserService {

	public List<User> getAllUsers();
	
	public Optional<User> getUserById(int userId);
	
	public Optional<User> newUser(User user);
	
	public User updateUser(int userId,  User user);
	
	public List<User> findUserByName(String name);
	
	public User findUserByEmail(String email);
	
	public void deleteUser(int userId);

	public User loginUser(User user);
	

}

File: frontend/src/app/user/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from '../models/user.model';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private API_URL = "http://localhost:8080/shoppingcart/users/";

  constructor(private http: HttpClient) { }


  signup(usarData: any): Observable<any> {
    return this.http.post(`${this.API_URL}add`,usarData)
  }

  getUserById(userId: number): Observable<User> {
    return this.http.get<User>(`${this.API_URL}${userId}`);
  }

  updateUser(userId: number, usarData: any): Observable<User> {
    return this.http.put<User>(`${this.API_URL}${userId}`,usarData)
  }

  // login(email: string, password: string): Observable<any> {
  //   return this.http.post<any>(`${this.API_URL}login`, {email, password})
  // }
}
